Sources/ProjectDescription/Project.swift

```diff
@@ -46,3 +50,71 @@ public struct Project: Codable, Equatable {
         dumpIfNeeded(self)
     }
 }
+
+// MARK: - Options
+
+extension Project {
+    /// Additional options related to the `Project`
+    public enum Options: Codable, Equatable {
+        /// Text settings to override user ones for currecnt project
+        case textSettings(TextSettings)
+    }
+}
+
+// MARK: - TextSettings
+
+extension Project.Options {
+    /// Text settings for Xcode project
+    public struct TextSettings: Codable, Equatable {
+        /// Use tabs over spaces
+        public let usesTabs: Bool?
+        /// Indent width
+        public let indentWidth: UInt?
+        /// Tab width
+        public let tabWidth: UInt?
+        /// Wrap lines
+        public let wrapsLines: Bool?
+        
+        public init(
+            usesTabs: Bool? = nil,
+            indentWidth: UInt? = nil,
+            tabWidth: UInt? = nil,
+            wrapsLines: Bool? = nil
+        ) {
+            self.usesTabs = usesTabs
+            self.indentWidth = indentWidth
+            self.tabWidth = tabWidth
+            self.wrapsLines = wrapsLines
+        }
+    }
+}
+
+// MARK: - Options + Codable
+
+extension Project.Options {
+    enum CodingKeys: String, CodingKey {
+        case textSettings
+    }
+
+    public init(from decoder: Decoder) throws {
```

Comment:
worth adding a sanity test for the Codable implementation seeing there's a custom one being used.

e.g.
https://github.com/tuist/tuist/blob/main/Tests/ProjectDescriptionTests/TargetTests.swift